sycl/test/abi/layout_vec.cpp

```diff
@@ -23,7 +22,6 @@ SYCL_EXTERNAL void foo(sycl::vec<bool, 16>) {}
 
 // CHECK:      0 | class sycl::vec<_Bool, 16>
 // ignore empty base classes
-// CHECK:      0 |   struct std::array<_Bool, 16> m_Data
-// CHECK-NEXT: 0 |     typename {{.+}}::_Type _M_elems
+// CHECK:      0 |   DataType m_Data
```

Comment:
Although this does not change the size of the `vec` type, I worry that this is breaking ABI as the `vec` type crosses the library boundary. To understand why this could be an issue, imagine the following scenario:

Let this change be part of a version Y of the runtime and headers, which came after a version X without it. By ABI/API guarantees we make, programs compiled with X should be runable with Y. However, if such a program passes `vec` across the library boundary, the runtime library of Y will think the data representation inside the type is a C-style array, while it is actually a `std::array`.

Granted, it's likely that C-style arrays and `std::array` are the exact same layout, but making such assumptions is also potentially playing with fire.

Reply:
I agree, by the definition given by CPPReference it _should_ (depending on how we should read "semantics") always have the same layout. That said, it gets awfully close to breaking strict-aliasing rules in a way you don't see every day.

Reply:
I see. To be on a safer side, I've put these changes under preview-breaking flag.